rma/models/rma.py

```diff
@@ -0,0 +1,903 @@
+# Copyright 2020 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tests import Form
+from odoo.tools import html2plaintext
+import odoo.addons.decimal_precision as dp
+from odoo.addons.stock.models.stock_move import PROCUREMENT_PRIORITIES
+
+
+class Rma(models.Model):
+    _name = "rma"
+    _description = "RMA"
+    _order = "priority,date desc"
+    _inherit = ["mail.thread", "portal.mixin", "mail.activity.mixin"]
+
+    # General fields
+    sent = fields.Boolean()
+    name = fields.Char(
+        string='Name',
+        index=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        copy=False,
+        default=lambda self: _('New'),
+    )
+    origin = fields.Char(
+        string='Source Document',
+        states={
+            'locked': [('readonly', True)],
+            'cancelled': [('readonly', True)],
+        },
+        help="Reference of the document that generated this RMA.",
+    )
+    date = fields.Datetime(
+        default=lambda self: fields.Datetime.now(),
+        index=True,
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    deadline = fields.Date(
+        states={
+            'locked': [('readonly', True)],
+            'cancelled': [('readonly', True)],
+        },
+    )
+    user_id = fields.Many2one(
+        comodel_name="res.users",
+        string="Responsible",
+        track_visibility="always",
+        default=lambda self: self.env.user,
+        states={
+            'locked': [('readonly', True)],
+            'cancelled': [('readonly', True)],
+        },
+    )
+    team_id = fields.Many2one(
```

Comment:
I'm missing to be able to decide where to receive the RMA. Maybe having the option to pick the operation (tha could be even a default for the team) would be the more direct way.